### ISSUE TITLE
"assembler: couldn't resolve label address" when compiling jumps or switches for t6.

### ISSUE THREAD

**Issue by INSANEMODE** [xensik/gsc-tool#124]
As an example, when using using `gsc-tool.exe comp t6 pc test\gsc-tool-1_0_667_for.gsc`, while the gsc contains a valid for loop,
will give you the following error:
 `[ERROR]:assembler: couldn't resolve label address of loc_3 at test/gsc-tool-1_0_667_for.gsc`

Seems to affect for, while, foreach, if, and switches.

Attaching sample scripts with each of the above that replicate the problem

Output from these files:
```
[ERROR]:assembler: couldn't resolve label address of loc_3 at test/gsc-tool-1_0_667_assembler_for.gsc
[ERROR]:assembler: couldn't resolve label address of loc_1 at test/gsc-tool-1_0_667_assembler_foreach.gsc
[ERROR]:assembler: couldn't resolve label address of loc_1 at test/gsc-tool-1_0_667_assembler_if.gsc
[ERROR]:assembler: couldn't resolve label address of loc_1 at test/gsc-tool-1_0_667_assembler_switch.gsc
[ERROR]:assembler: couldn't resolve label address of loc_1 at test/gsc-tool-1_0_667_assembler_while.gsc
```

Other modes and systems have not been attempted to see if they have the same outcome.

When attempting to debug, an assert fails while in the for loop in assemble_function https://github.com/xensik/gsc-tool/blob/5a144a3ba7d7d53adc6bc74ce96f1668cb5941ae/src/arc/assembler.cpp#L229

![image](https://github.com/xensik/gsc-tool/assets/10606102/8eef37d3-cde4-4976-a5fe-988758e57bf9)

[test.zip](https://github.com/xensik/gsc-tool/files/11514862/test.zip)

**Comment by xensik** [comment]
i cant reproduce, your files compile correctly on 1.0.667 for me
<img width="444" alt="Captura de Pantalla 2023-05-19 a las 12 26 01" src="https://github.com/xensik/gsc-tool/assets/8714336/b23b3438-3507-4eaf-85ab-534018ad8a7c">

**Comment by xensik** [comment]
seems to be a problem on windows binaries

**Comment by xensik** [comment]
fixed #125
